cli/flow-typed/jest.js.flow

```diff
@@ -3,10 +3,55 @@ declare function describe(suiteName: string, cb: () => void): void;
 declare function expect(value: any): Object;
 declare function it(testName: string, cb: (done: () => void) => void): void;
 declare function pit(testName: string, cb: () => Promise<void>): void;
+
+type MockFn = {
+  (...args: Array<any>): any;
+  mock: {
+    calls: Array<Array<any>>;
+    instances: any;
+  };
+  mockClear(): Function;
+  mockImplementation(fn: Function): MockFn;
+  mockReturnThis(): any;
+  mockReturnValue(value: any): MockFn;
+  mockReturnValueOne(value: any): MockFn;
+}
+
+type ExpectType = {
+  not: ExpectType;
+  toThrow(message?: string): void;
+  toBe(value: any): void;
+  toEqual(value: any): void;
+  toBeFalsy(): void;
+  toBeTruthy(): void;
+  toBeNull(): void;
+  toBeUndefined(): void;
+  toBeDefined(): void;
+  toMatch(regexp: RegExp): void;
+  toContain(str: string): void;
+  toBeCloseTo(num: number, delta: any): void;
+  toBeGreaterThan(number: number): void;
+  toBeLessThan(number: number): void;
+  toBeCalled(): void;
+  toBeCalledWith(...args: Array<any>): void;
+  lastCalledWith(...args: Array<any>): void;
+}
+
+declare function expect(value: any): ExpectType;
+
 declare var jest: {
-  disableAutomock: () => void,
-  enableAutomock: () => void,
-  fn: <T>(impl?: T) => T,
-  mock: (moduleName: string, impl?: () => any) => void,
-  unmock: (moduleName: string) => void,
-};
+  genMockFunction(): MockFn;
```

Comment:
let's get rid of this, we now have `jest.fn` for this and `genMockFunction` is deprecated.